### PR TITLE
Change NNCF part of templates

### PR DESCRIPTION
**Pull request by LeonidBeynenson**
* Make NNCF configuration in template files be compliant with the external specs
* Add possibility to turn on NNCF compression flags from command line arguments of ote tools

### CODE REVIEW

**Comment by LeonidBeynenson** [comment]
2 @Ilya-Krylov :
Ilya, I made all the considered changes, resolved merge conflicts,  and ran tests on my desktop.
As far as I can see all tests are passed (see the attachment).

[desktop_run_all_tests__2020-12-19_00-28-19.zip](https://github.com/openvinotoolkit/training_extensions/files/5718591/desktop_run_all_tests__2020-12-19_00-28-19.zip)

**Comment by LeonidBeynenson** [comment]
run tests

File: pytorch_toolkit/ote/tools/train.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from ote import MODEL_TEMPLATE_FILENAME, MODULES_CONFIG_FILENAME
 from ote.utils import load_config
 from ote.modules import build_arg_parser, build_arg_converter, build_trainer, build_compression_arg_transformer
-from ote.modules.compression import is_compression_enabled_in_template
+from ote.modules.compression import is_optimisation_enabled_in_template
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/tools/train.py:22]: is not used anymore

File: pytorch_toolkit/ote/ote/tests/test_case.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 
 from ote.tests.utils import collect_ap, run_through_shell
 from ote.utils.misc import download_snapshot_if_not_yet
+from ote.modules.compression import (is_optimisation_enabled_in_template,
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/tests/test_case.py:27]: sort imports please

File: pytorch_toolkit/ote/ote/modules/exporters/mmdetection.py [hunk 3]
```diff
@@ -40,6 +49,12 @@ def _export_to_openvino(self, args, tools_dir):
         should_run_alt_ssd_export = (hasattr(config.model, 'bbox_head')
                                      and config.model.bbox_head.type == 'SSDHead'
                                      and not config.get('nncf_config'))
+
+        if is_checkpoint_nncf and is_checkpoint_nncf(args['load_weights']):
+            # if the config does not contain NNCF part,
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/modules/exporters/mmdetection.py:54]: how does this comment correlate to the code

File: pytorch_toolkit/ote/ote/modules/compression/nncf_config_generator.py [hunk 4]
```diff
@@ -25,94 +25,118 @@
 
 # TODO(LeonidBeynenson): implement unit tests on NNCFConfigGenerator
 
-POSSIBLE_NNCF_PARTS = {'int8', 'sparsity', 'pruning'}
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/modules/compression/nncf_config_generator.py:28]: Have you implemented the tests? See two lines above, if yes please remove TODO

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

File: pytorch_toolkit/ote/tools/train.py (hunk 1 above)
Comment (LeonidBeynenson) [reply pytorch_toolkit/ote/tools/train.py:22]: Ok

Comment (LeonidBeynenson) [reply pytorch_toolkit/ote/tools/train.py]: Fixed

File: pytorch_toolkit/ote/ote/tests/test_case.py (hunk 2 above)
Comment (LeonidBeynenson) [reply pytorch_toolkit/ote/ote/tests/test_case.py]: Fixed

File: pytorch_toolkit/ote/ote/modules/exporters/mmdetection.py (hunk 3 above)
Comment (LeonidBeynenson) [reply pytorch_toolkit/ote/ote/modules/exporters/mmdetection.py]: Fixed the comment -- made it more descriptive

File: pytorch_toolkit/ote/ote/modules/compression/nncf_config_generator.py (hunk 4 above)
Comment (LeonidBeynenson) [reply pytorch_toolkit/ote/ote/modules/compression/nncf_config_generator.py:28]: No, I have not implemented them yet.
So, the comment it still required

File: pytorch_toolkit/ote/ote/tests/test_case.py [hunk 5]
```diff
@@ -24,8 +24,8 @@
 
 from ote.tests.utils import collect_ap, run_through_shell
 from ote.utils.misc import download_snapshot_if_not_yet
-from ote.modules.compression import (is_optimisation_enabled_in_template,
-                                     get_optimisation_config_from_template)
+from ote.modules.compression import (get_optimisation_config_from_template,
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/tests/test_case.py:27]: On higher level)

Comment (LeonidBeynenson) [reply pytorch_toolkit/ote/ote/tests/test_case.py:25]: Ooooops....
Fixed!

**Review by Ilya-Krylov — APPROVED** [review]

**Pull request merged**